interface/patient_file/encounter/Services/ViewHelper.php

```diff
@@ -0,0 +1,274 @@
+<?php
+/**
+ * Helper functions for an encounter view files
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package OpenEMR
+ * @subpackage Encounter
+ * @author Robert Down <robertdown@live.com>
+ * @copyright Copyright (C) 2017 Robert Down
+ */
+
+namespace OpenEMR\Encounter\Services;
+
+class ViewHelper
+{
+
+    private $oldCategory = '';
+
+    static function getRegistry($state = "1", $limit = "unlimited", $offset = "0")
+    {
+        global $attendant_type;
+        $sql = "SELECT * FROM registry WHERE {column}= 1 AND state LIKE ? ORDER BY category, priority, name";
+        $params = array();
+        $column = ($attendant_type == 'pid') ? 'patient_encounter' : 'therapy_group_encounter';
+        $sql = str_replace("{column}", $column, $sql);
```

Comment:
1. Where is the $column variable coming from?
2. Need to escape $column via whitelisting technique:
``````
$sql = str_replace("{column}", escape_sql_column_name($column,array("registry")), $sql);